File: Backend/schemas/service_price_history.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class ServicePriceHistoryResponse(BaseModel):
    id: int
    service_id: int
    old_price: float
    new_price: float
    updated_by: Optional[int] = None
    updated_at: datetime

    class Config:
        from_attributes = True

File: flutter_application_1/lib/models/service_price_history.dart
class ServicePriceHistory {
  final String id;
  final String serviceId;
  final double oldPrice;
  final double newPrice;
  final String updatedBy;
  final DateTime updatedAt;

  ServicePriceHistory({
    required this.id,
    required this.serviceId,
    required this.oldPrice,
    required this.newPrice,
    required this.updatedBy,
    required this.updatedAt,
  });

  factory ServicePriceHistory.fromJson(Map<String, dynamic> json) =>
      ServicePriceHistory(
        id: json['id']?.toString() ?? '',
        serviceId: json['service_id']?.toString() ?? '',
        oldPrice: (json['old_price'] as num).toDouble(),
        newPrice: (json['new_price'] as num).toDouble(),
        updatedBy: json['updated_by']?.toString() ?? '',
        updatedAt: DateTime.parse(json['updated_at']),
      );
}
